packages/api-page-builder-import-export/src/import/utils/uploadAssets.ts

```diff
@@ -28,13 +28,19 @@ export const uploadAssets = async (params: UploadAssetsParams) => {
         return oldIdToNewFileMap;
     }
 
+    // Check if files with such keys already exist
+    const [existingFiles] = await context.fileManager.files.listFiles();
+    const filteredFiles = files.filter(
+        file => !existingFiles.some(existingFile => existingFile.key === file.key)
+    );
+
```

Comment:
This will not work; this means you're potentially loading millions of files. One of our clients now is importing 3 millions of images, so you can imagine how this will not end well :D 

When you're exporting a page, let's say it has 20 images, you know the exact image IDs, so you can use the `ids` parameter on listFiles, like this:


```
const existingImages = await context.fileManager.files.listFiles({ where: { id_in: existinIds} })
```